File: src/main/java/com/lcaohoanq/nocket/domain/otp/IOtpService.java
package com.lcaohoanq.nocket.domain.otp;

import java.util.Optional;
import java.util.UUID;

public interface IOtpService {

    Otp createOtp(Otp otp);
    void disableOtp(UUID id);
    Optional<Otp> getOtpByEmailOtp(String email, String otp);
    void setOtpExpired();
}

File: src/main/java/com/lcaohoanq/nocket/domain/otp/IOtpService.kt
package com.lcaohoanq.nocket.domain.otp

import java.util.*

interface IOtpService {
    fun createOtp(otp: Otp): Otp
    fun disableOtp(id: UUID)
    fun getOtpByEmailOtp(email: String, otp: String): Optional<Otp>
    fun setOtpExpired()
    fun generateOtp(): String
}
